### ISSUE TITLE
Question about paper

### ISSUE THREAD

**Issue by TwjLoveCode** [thu-coai/PPT#9]
Hi, author!
In paper, it writes:
"Note that in this way, the pre-trained PVPs can be used in single text classification tasks from arbitrary domains and with much more labels."(In 3.3)
"For tasks with more labels, especially single-text classification where pseudo label pre-training is not appropriate for cross-domain adaption, Unified PPT is a good alternative."(In 4.2)

and i wonder, is PPT a another good way for domain-adaptation issue, or is it practical for DA? And, if i am faced with a normal cross-domain sentiment analysis（2 label: good or bad）,is PPT useful? Before i try it, i am here to ask you for some advice :)

THANK YOU!!!

**Comment by t1101675** [comment]
Hi, thanks for your interest to our work! PPT is not originally designed for DA. But I think Unified-PPT can be a good way for cross-domain sentiment analysis because in Unified-PPT, we pre-train the soft prompts on the corpus from a variety of sources and domains, which helps the model generalization across different domains, especially when your training data is scarce.

**Comment by TwjLoveCode** [comment]
thanks for your answering. And, is it possible that using your feamework but based on a much smaller model such as RoBERTa.. etc?if so, will it still perform well and how to change model without changing your framework

**Comment by t1101675** [comment]
I think it's possible to transform it to RoBERTa model. Compared to vanilla prompt tuning based on RobERTa, PPT should still achieve better performance. 
However, the gap between **full-model fine-tuning** and **vanilla prompt tuning** will be larger than the large model setting. Therefore, I'm not sure whether PPT can still be comparable to full-model fine-tuning.
If you want to change the model, you may need to change the [model definition](https://github.com/thu-coai/PPT/blob/master/model/enc_dec_modeling.py) from T5 to RoBERTa . There  are maybe also some details need to fix in the [transformer blocks](https://github.com/thu-coai/PPT/blob/master/mpu/transformer_enc_dec.py) like the pre/post layer-norm. The model definition mostly follow the Huggingface's implement.

**Comment by t1101675** [comment]
If you find changing the model difficult, I recommend to fist try smaller version of T5 model, which is much more easier to test in our framework. You only need to change the configuration file in the training scripts to [smaller-model configurations](https://github.com/thu-coai/PPT/tree/master/configs/model).
